shell-ui/src/auth/permissionUtils.js

```diff
@@ -0,0 +1,40 @@
+//@flow
+import type { Options, TranslationAndGroups } from '../index';
+
+export const isEntryAccessibleByTheUser = (
+  [path, translationAndGroup]: [string, TranslationAndGroups],
+  userGroups: string[],
+): boolean => {
+  return (
+    (translationAndGroup.groups &&
+      translationAndGroup.groups
+        .map((group) => userGroups.includes(group))
+        .reduce(
+          (aggregatedPredicates, current) => aggregatedPredicates && current,
+          true,
+        )) ||
+    !translationAndGroup.groups
+  );
```

Comment:
Could be written more simply with `Array.prototype.every`:

```suggestion
  return (
    translationAndGroup.groups?.every(
      (group) => userGroups.includes(group)
    ) || true
  );
```

(one nice benefit of `every` is that it will automatically return `true` if your array is empty :wink:)